django/template/base.py

```diff
@@ -405,7 +405,7 @@ def tokenize(self):
 
 class Parser:
     def __init__(self, tokens, libraries=None, builtins=None, origin=None):
-        self.tokens = tokens
+        self.tokens = list(reversed(tokens))
```

Comment:
Perhaps it'd help to have a comment here explaining why we do this, in order to avoid confusing people who read it later?

I might have done `self.tokens = tokens[::-1]`, but I doubt it makes much difference.

Reply:
Done